src/textual/view.py

```diff
@@ -114,7 +114,24 @@ def get_arrangement(self, size: Size, scroll: Offset) -> Iterable[WidgetPlacemen
         cached_size, cached_scroll, arrangement = self._cached_arrangement
         if cached_size == size and cached_scroll == scroll:
             return arrangement
-        arrangement = list(self._layout.arrange(self, size, scroll))
+
+        arrangement = []
+        placements = self._layout.arrange(self, size, scroll)
+        for placement in placements:
+            region, widget, order = placement
```

Comment:
Could you do this within the loop, i.e. `for region, widget, order in placements`.